### PR TITLE
feat: fetch landlords sharehouses

### LINKED ISSUE #250: Fetch landlord's sharehouses
## Description

Please fetch data from `/api/sharehouses` to list all the sharehouses registered by the landlord.

## Criteria

- [ ] Fetch and display data from `/api/sharehouses`
- [ ] If applicable, delete hard-coded code like mock data or related code

### PR DESCRIPTION
**Pull request by HrRn**
## Overview

- I have added fetching of the landlord's sharehouses.

## Changes

- Call the `api.sharehouses.$get` method to fetch share house information.
- Modify the share house's maximum amount based on the provided code.
- Update the shareHouse props.

## Screenshots or videos

sharehouses

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [x] Understandable and maintainable code

### CODE REVIEW

File: src/app/sharehouses/page.tsx [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
-    },
-  ];
+  );
+
+  // Convert response to JSON
+  const data = await res.json();
+
+  // Check for error in data and display it if found
+  if ('error' in data) {
+    return <div>{data.error}</div>;
```
Comment (akiomatic) [inline src/app/sharehouses/page.tsx:26]: This `<div>` is temporary, right? Then, could you add a TODO comment so that it can be noticed to fix it later?

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you! I've left a comment.

File: src/app/sharehouses/page.tsx (hunk 1 above)
Comment (HrRn) [reply src/app/sharehouses/page.tsx:26]: Thank you for reviewing! I added a TODO comment.

**Review by akiomatic — APPROVED** [review]
LGTM. Thank you!

**Review by Akikaze1119 — APPROVED** [review]
LGTM! Thank you :)

**Review by yuki-92c — APPROVED** [review]
LGTM, Thanks so much

**Pull request merged**